evalml/utils/gen_utils.py

```diff
@@ -219,8 +219,16 @@ def _rename_column_names_to_numeric(X):
     Returns:
         Transformed X where column names are renamed to numerical values
     """
-    name_to_col_num = dict((col, col_num) for col_num, col in enumerate(X.columns.values))
-    return X.rename(columns=name_to_col_num, inplace=False)
+    X_t = X
+    if isinstance(X, np.ndarray):
+        return X
```

Comment:
Should we normalize to dataframe here too? Or we're assuming the caller does that?

Reply:
Ah, you're right, we should normalize here!